tests/nn/data_parallel/test_fsdp_input.py

```diff
@@ -28,53 +40,45 @@
     "fsdp_config", [{}, {"mixed_precision": True}],
 )
 @pytest.mark.parametrize("input_cls", [dict, list])
-def test_it(fsdp_config, input_cls):
+def test_input_type(temp_files, fsdp_config, input_cls):
     """Test FSDP with input being a list or a dict, only single GPU."""
-    if torch_version() < (1, 6, 0):
-        pytest.skip("older pytorch doesn't support reduce_scatter")
-
-    # Random port in case the next test run quickly, same port would cause conflict.
-    os.environ["MASTER_ADDR"] = "localhost"
-    os.environ["MASTER_PORT"] = str(random.randint(2000, 3000))
-    torch.distributed.init_process_group(backend="nccl", rank=0, world_size=1)
-
-    try:
-        assert isinstance(fsdp_config, dict), str(fsdp_config)
-
-        class Model(Module):
-            def __init__(self):
-                super().__init__()
-                self.layer = Linear(4, 4)
-
-            def forward(self, input):
-                if isinstance(input, list):
-                    input = input[0]
-                else:
-                    assert isinstance(input, dict), input
-                    input = input["in"]
-                return self.layer(input)
-
-        model = FSDP(Model(), **fsdp_config).cuda()
-        optim = SGD(model.parameters(), lr=0.1)
-
-        for _ in range(5):
-            in_data = torch.rand(64, 4).cuda()
-            in_data.requires_grad = True
-            if input_cls is list:
-                in_data = [in_data]
-            else:
-                assert input_cls is dict
-                in_data = {"in": in_data}
 
-            out = model(in_data)
-            out.sum().backward()
-            optim.step()
-            optim.zero_grad()
+    world_size = 1
+    result = dist_init(0, world_size, temp_files[0], temp_files[1])
+    assert result, "Dist init failed"
+
+    assert isinstance(fsdp_config, dict), str(fsdp_config)
 
-        model.assert_state(TrainingState.IDLE)
+    class Model(Module):
+        def __init__(self):
+            super().__init__()
+            self.layer = Linear(4, 4)
 
-    finally:
-        # Clean-up is important or the next test in this file may fail to init the PG.
-        torch.distributed.destroy_process_group()
-        del os.environ["MASTER_ADDR"]
-        del os.environ["MASTER_PORT"]
+        def forward(self, input):
+            if isinstance(input, list):
+                input = input[0]
+            else:
+                assert isinstance(input, dict), input
+                input = input["in"]
+            return self.layer(input)
+
+    model = FSDP(Model(), **fsdp_config).cuda()
+    optim = SGD(model.parameters(), lr=0.1)
+
+    for _ in range(5):
```

Comment:
why do we need to do this 5 times? why not just once?

Reply:
main reason is to (1) cover the difference between 1st iter and 2nd iter and (2) run a few more so let error accumulate a bit to trigger numerical mismatch error -- too few iters may not be enough to detect errors. It is ad hoc.

Reply:
Ah I see. Maybe I missed it but I did not see us asserting on anything other than input type so I thought a single iteration would be sufficient.

Reply:
You are right, for this particular test, 5 is likely not needed. Must be a copy paste from somewhere else. Since it is very fast already, mind I just leave as is?